packages/core/src/components/tabs2/tabs2.tsx

```diff
@@ -153,6 +153,12 @@ export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
     public componentDidUpdate(_prevProps: ITabs2Props, prevState: ITabs2State) {
         if (this.state.selectedTabId !== prevState.selectedTabId) {
             this.moveSelectionIndicator();
+        } else if (prevState.selectedTabId !== undefined && prevState.selectedTabId !== null) {
```

Comment:
`prevState.selectedTabId != null` (double-not-equals) handles both of these values.